kore/src/Kore/Equation/Application.hs

```diff
@@ -176,6 +210,40 @@ attemptEquation sideCondition termLike equation =
         debugAttemptEquationResult equation result
         return result
 
+    simplifyArgumentWithResult
+        :: MonadTrans t
+        => Predicate (Target variable)
+        -> Maybe (Predicate (Target variable))
+        -> MatchResult (Target variable)
+        -> t simplifier [MatchResult (Target variable)]
+    simplifyArgumentWithResult
+        argument
+        antiLeft
+        (matchPredicate, matchSubstitution)
+      =
+        -- TODO: toMap is unsafe;
+        -- is it impossible to happen (if the frontend generates equations
+        -- correctly), or should we add a new AttemptEquationError case?
```

Comment:
Yes, this is impossible, but let's add a `HasCallStack` constraint so that we get a reasonable message if it ever fails.